public/app/features/datasources/DataSourcesListItem.tsx

```diff
@@ -1,36 +1,240 @@
-import React, { PureComponent } from 'react';
+import React, { FunctionComponent, useState, useContext } from 'react';
 import { DataSourceSettings } from '@grafana/data';
 import { e2e } from '@grafana/e2e';
+import appEvents from 'app/core/app_events';
+import { CoreEvents } from 'app/types/';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { stylesFactory, ThemeContext } from '@grafana/ui';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  dataSourceCard: css`
+    display: table-row;
+    cursor: pointer;
+    background: ${theme.background.panelEditorVizItem};
+
+    &:hover {
+      background: ${theme.background.panelEditorVizItemHover};
+      color: ${theme.colors.textStrong};
+    }
+  `,
+  dataSourceLogoWrapper: css`
+    display: table-cell;
+    text-align: center;
+    vertical-align: middle;
+    padding: ${theme.spacing.md};
+
+    border-width: 2px 0 2px 2px;
+    border-style: solid;
+    border-color: ${theme.colors.primaryBorder};
+    border-radius: 3px 0 0 3px;
+  `,
+  dataSourceLogoWrapperOnCardHover: css`
+    border-color: ${theme.colors.primaryBorderHover};
+  `,
+  dataSourceLogo: css`
+    min-width: 55px;
+    width: 55px;
+    min-height: 55px;
+    max-height: 55px;
+    margin-bottom: ${theme.spacing.xxs};
+  `,
+  dataSourceNameWrapper: css`
+    display: table-cell;
+    vertical-align: middle;
+    border-width: 2px 0;
+    border-style: solid;
+    border-color: ${theme.colors.primaryBorder};
+  `,
+  dataSourceNameWrapperOnCardHover: css`
+    border-color: ${theme.colors.primaryBorderHover};
+  `,
+  dataSourceName: css`
+    font-size: ${theme.typography.heading.h5};
+    display: block;
+  `,
+  dataSourceType: css`
+    font-size: ${theme.typography.size.sm};
+    color: ${theme.colors.textWeak};
+    display: block;
+  `,
+  dataSourceURLWrapper: css`
+    border-width: 2px 0;
+    border-style: solid;
+    border-color: ${theme.colors.primaryBorder};
+  `,
+  dataSourceURLWrapperOnCardHover: css`
+    border-color: ${theme.colors.primaryBorderHover};
+  `,
+  dataSourceURL: css`
+    font-size: ${theme.typography.heading.h5};
+  `,
+  dataSourceActionsWrapper: css`
+    text-align: center;
+    min-width: 240px;
+    border-width: 2px 2px 2px 0;
+    border-style: solid;
+    border-color: ${theme.colors.primaryBorder};
+    border-radius: 0 3px 3px 0;
+  `,
+  dataSourceActionsWrapperOnDelete: css`
+    text-align: left;
+  `,
+  dataSourceActionsWrapperOnCardHover: css`
+    opacity: 1;
+    transition: 0.15s opacity ease-in-out;
+    border-color: ${theme.colors.primaryBorderHover};
+  `,
+  actionsLink: css`
+    opacity: 0;
+    height: 16px;
+    margin: 0 8px;
+    vertical-align: middle;
+
+    &:before {
+      font-size: 16px;
+      position: relative;
+    }
+  `,
+  actionsLinkOnCardHover: css`
+    opacity: 0.6;
+
+    &:hover {
+      opacity: 1;
+      transition: 0.15s opacity ease-in-out;
+    }
+  `,
```

Comment:
This looks strange, have a hover class with a hover rule under it? can't all the hover state changes be added to a single `&:hover` nested rule on the card item class? They increased text brighness (using textStrong color instead of changing opacity)